src/v/storage/disk_log_impl.cc

```diff
@@ -474,8 +475,9 @@ ss::future<compaction_result> disk_log_impl::compact_adjacent_segments(
     // log operations are later identified before committing changes.
     auto staging_path = std::filesystem::path(
       fmt::format("{}.compaction.staging", target->reader().filename()));
-    auto replacement = co_await storage::internal::make_concatenated_segment(
-      staging_path, segments, cfg, _manager.resources());
+    auto [replacement, generations]
+      = co_await storage::internal::make_concatenated_segment(
```

Comment:
Should we set the generation of the resulting compacted segment to generation of the first segment + 1? So that we don't confuse it with the old first segment.

Reply:
Discussed offline that as we don't re-use in-memory segment objects, this is not needed.